karate-tests/src/test/java/patients/healthcareWorker/postPatient.feature

```diff
@@ -1,48 +1,138 @@
 Feature: Create a patient - Healthcare worker access mode
 
 Background:
-    * def utils = call read('classpath:helpers/utils.feature')
-    * def accessToken = karate.callSingle('classpath:patients/healthcareWorker/auth-redirect.feature').accessToken
-    * def requestHeaders = call read('classpath:patients/healthcareWorker/healthcare-worker-headers.js')
-    * configure headers = requestHeaders  
-    * url baseURL
-
-Scenario: Post patient, required request data only
-    * def familyName = 'Karate-test-' + utils.randomInt()
-    * def body =
-    """
-    {
-        "nhsNumberAllocation": "Done",
-        "name": {
-            "use": "L",
-            "name.familyName": "#(familyName)"
-        },
-        "registeringAuthority": {
-            "regAuthorityType.code": "x",
-            "regAuthorityType.codeSystem": "2.16.840.1.113883.2.1.3.2.4.16.20",
-            "regOrganisation.root": "2.16.840.1.113883.2.1.4.3",
-            "regOrganisation.extension": "RWF",
-            "authorPersonID": "",
-            "authorSystemID": "230811201324",
-            "deathStatus": "",
-            "deceasedTime": "",
-            "overallUpdateMode": "create"
-        }
-    }
-    """
-    * path "Patient"
-    * request body
-    * method post
-    * status 201
-    * def nhsNumber = response.id
-    * def expectedResponse = read('classpath:stubs/patient/new-nhs-number-response-template.json')
-    * match response == expectedResponse
+  * def utils = call read('classpath:helpers/utils.feature')
+  * def faker = Java.type('helpers.FakerWrapper')
+  
+  # the same registeringAuthority object is included in every request
+  * def registeringAuthority = 
+  """
+  {
+    "regAuthorityType.code": "x",
+    "regAuthorityType.codeSystem": "2.16.840.1.113883.2.1.3.2.4.16.20",
+    "regOrganisation.root": "2.16.840.1.113883.2.1.4.3",
+    "regOrganisation.extension": "RWF",
+    "authorPersonID": "",
+    "authorSystemID": "230811201324",
+    "deathStatus": "",
+    "deceasedTime": "",
+    "overallUpdateMode": "create"
+  }
+  """
+
+  # the addressSchema used for the create responses is not the same as what's defined elsewhere, so 
+  # we'll use this schema until the schemas are harmonised...
+  * def addressSchema = 
+  """
+  {
+    "id": "#regex([0-9A-Z]+)",
+    "line": [
+        "#string"
+    ],
+    "period": {
+        "start": "#? utils.isTodaysDate(_)"
+    },
+    "postalCode": "#? utils.isValidPostalCode(_)",
+    "use": "#regex(home)"
+  }
+  """
+  
+  * def accessToken = karate.callSingle('classpath:patients/healthcareWorker/auth-redirect.feature').accessToken
+  * def requestHeaders = call read('classpath:patients/healthcareWorker/healthcare-worker-headers.js')
+  * configure headers = requestHeaders  
+  * url baseURL
+
+  # we have to chill out a bit between requests in case we trigger the spike alert policy :-(
+  * eval utils.sleep(4)
```

Comment:
Interesting this needs to be 4 seconds? I would have thought limiting it to 1 a second would be more than enough. 🤔

Reply:
You would've thought.... I'm surprised at how fast this spike alert policy is triggered tbh. I started off low, but I was still getting spike alerts, so I went higher until I got consistent passes.

Reply:
Might be worthwhile seeing what the traces look like in Apigee when this step runs in the pipeline. :shrug:

Reply:
Got a much nicer solution using karate retry functionality

```gherkin
  * configure retry = { count: 5, interval: 1 }
  * retry until responseStatus != 429
  * method post
 ```